### ISSUE TITLE
VS code crashes when debugging after update with Node <=12

### ISSUE THREAD

**Issue by iii-james** [microsoft/vscode-js-debug#1624]
**Describe the bug**
After VS code updated to 1.77.0 I can no longer debug my NodeJS application.

**To Reproduce**
Click the debug button

**Launch.json**
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Program",
            "program": "${workspaceRoot}/server/src/index.ts",
            "sourceMaps": true,
            "outFiles": [
                "${workspaceRoot}/server/dist/**/*.js"  
            ]
        }
    ]
}
```
**Debug Console**
```
        c:\Users\James.Morris\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\ms-vscode.js-debug\src\bootloader.js:8
        `+r)),e.removeListener("error",er),e.destroy()}});var Rs=V((Vu,Cs)=>{"use strict";var xa=Object.create,pt=Object.defineProperty,ba=Object.getOwnPropertyDescriptor,Ss=Object.getOwnPropertyNames,va=Object.getPrototypeOf,Sa=Object.prototype.hasOwnProperty,Ea=(e,t)=>function(){return t||(0,e[Ss(e)[0]])((t={exports:{}}).exports,t),t.exports},Ta=(e,t)=>{for(var r in t)pt(e,r,{get:t[r],enumerable:!0})},Es=(e,t,r,n)=>{if(t&&typeof t=="object"||typeof t=="function")for(let s of Ss(t))!Sa.call(e,s)&&s!==r&&pt(e,s,{get:()=>t[s],enumerable:!(n=ba(t,s))||n.enumerable});return e},Ca=(e,t,r)=>(r=e!=null?xa(va(e)):{},Es(t||!e||!e.__esModule?pt(r,"default",{value:e,enumerable:!0}):r,e)),Ra=e=>Es(pt({},"__esModule",{value:!0}),e),Aa=Ea({"node_modules/request-light/lib/node/main.js"(e){(()=>{var t={46:(i,a)=>{"use strict";Object.defineProperty(a,"__esModule",{value:!0}),a.default=function(l,f,{signal:
        
        SyntaxError: Unexpected token .
            at Module._compile (internal/modules/cjs/loader.js:723:23)
            at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
            at Module.load (internal/modules/cjs/loader.js:653:32)
            at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
            at Function.Module._load (internal/modules/cjs/loader.js:585:3)
            at Module.require (internal/modules/cjs/loader.js:692:17)
            at require (internal/modules/cjs/helpers.js:25:18)
            at Object.<anonymous> (C:\Users\JAMES~1.MOR\AppData\Local\Temp\vscode-js-debug-bootloader.js:1:1)
            at Module._compile (internal/modules/cjs/loader.js:778:30)
            at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
        Process exited with code 1
```

**Log File**
[vscode-debugadapter-c0d5382f.json.gz](https://github.com/microsoft/vscode-js-debug/files/11115447/vscode-debugadapter-c0d5382f.json.gz)

**VS Code Version:** 
Version: 1.77.0 (user setup)
Commit: 7f329fe6c66b0f86ae1574c2911b681ad5a45d63
Date: 2023-03-29T10:02:16.981Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No

**Additional context**

**Comment by connor4312** [comment]
Can you grab a recording/gif of your vscode window? https://gifcap.dev/ is helpful

**Comment by iii-james** [comment]
![Recording 2023-03-30 at 15 50 16](https://user-images.githubusercontent.com/46497848/228961411-01dd38a6-2097-46be-811e-a321cf083df2.gif)

**Comment by connor4312** [comment]
Thanks, looks like that's an issue on very old Node.js versions. I will put in a patch for this, but I also strongly recommend upgrading as your version of Node.js has been [end of life](https://endoflife.date/nodejs) for about two years now and no longer receives security updates.

**Comment by iii-james** [comment]
@connor4312 Thanks for looking into this. I plan on updating NodeJS later. I have some dependency issues to work out first. Is there a way I can track the progress of the patch?

**Comment by connor4312** [comment]
I'll put a fix in for this today. When the issue is closed, it'll be fixed in sources. You can expect it to appear in the [nightly build](https://github.com/microsoft/vscode-js-debug#nightly-extension) shortly after 5PM PST the same day.

**Comment by connor4312** [comment]
This will be fixed in the [nightly build](https://github.com/microsoft/vscode-js-debug#nightly-extension) available shortly after 5PM PST today. I will also propose it for a recovery release next week.

**Comment by qdevico** [comment]
> 

Still not working, but now without crashing and errors

**Comment by duzenko** [comment]
It just freezes

**Comment by connor4312** [comment]
Thanks for letting me know, it looks like esbuild by default can change syntax during its minification process, which I didn't catch when developing the (unminified) extension locally. New build incoming in a little while.

**Comment by starball5** [comment]
Related question on Stack Overflow: [SyntaxError: Unexpected token '.' along with /ms-vscode.js-debug/src/bootloader.js:8](https://stackoverflow.com/q/75921266/11107541).

**Comment by rafaelpalladino** [comment]
@connor4312 how do I get the fix? I need to wait for a future VS Code release next week?

**Comment by connor4312** [comment]
I just published the new nightly build with that fix, version `v2023.4.312`. It'll take 5-10 minutes to propagate to the CDN.

If you don't see this, run "Check for extension updates", and a reload button should appear on the nightly extension build
<img width="585" alt="image" src="https://user-images.githubusercontent.com/2230985/229604056-e22859e8-96b4-4cef-8959-a579e0b4e0a2.png">

**Comment by rebornix** [comment]
Verified and it started the debugger successfully

```
Note: Using the "preview" debug extension
/Users/penlv/.nvm/versions/node/v10.24.1/bin/node ./test.js
```
